File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/token/SecurityTokenStore.java
package pers.acp.admin.oauth.token;

import org.springframework.security.oauth2.provider.token.TokenStore;
import pers.acp.admin.oauth.vo.LoginLogVO;

import java.util.List;

/**
 * @author zhang by 08/03/2019
 * @since JDK 11
 */
public interface SecurityTokenStore extends TokenStore {

    /**
     * 记录获取token的次数
     *
     * @param appId  应用id
     * @param userId 用户id
     */
    void storeLoginNum(String appId, String userId) throws Exception;

    /**
     * 获取token的次数
     *
     * @param appId 应用id
     * @return 次数
     */
    List<LoginLogVO> getLoginNum(String appId) throws Exception;

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/token/SecurityTokenStore.kt
package pers.acp.admin.oauth.token

import org.springframework.security.oauth2.provider.token.TokenStore
import pers.acp.admin.oauth.vo.LoginLogVo

/**
 * @author zhang by 08/03/2019
 * @since JDK 11
 */
interface SecurityTokenStore : TokenStore {

    /**
     * 记录获取token的次数
     *
     * @param appId  应用id
     * @param userId 用户id
     */
    @Throws(Exception::class)
    fun storeLoginNum(appId: String, userId: String)

    /**
     * 获取token的次数
     *
     * @param appId 应用id
     * @return 次数
     */
    @Throws(Exception::class)
    fun getLoginNum(appId: String): MutableList<LoginLogVo>

}
